tests/unit/test_braille.py

```diff
@@ -0,0 +1,96 @@
+#tests/unit/test_braille.py
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2017 NV Access Limited, Babbage B.V.
+
+"""Unit tests for the braille module.
+"""
+
+import unittest
+import braille
+from . import PlaceholderNVDAObject
+import controlTypes
+import api
+from config import conf
+import api
+import globalVars
+
+class NVDAObjectWithRole(PlaceholderNVDAObject):
+	"""An object that accepts a role as one of its construction parameters.
+	The name of the object will be set with the associated role label.
+	This class is used to quickly create objects for a fake focus ancestry."""
+
+	def __init__(self, role=controlTypes.ROLE_UNKNOWN,**kwargs):
+		super(NVDAObjectWithRole,self).__init__(**kwargs)
+		self.role=role
+
+	def _get_name(self):
+		return controlTypes.roleLabels.get(self.role,controlTypes.ROLE_UNKNOWN)
+
+class TestFocusContextPresentation(unittest.TestCase):
+	"""A test for the different focus context presentation options."""
+
+	@property
+	def regionsWithPositions(self):
+		return list(braille.handler.buffer.regionsWithPositions)
+
+	def setUp(self):
+		"""Set up a fake focus object and give it some ancestry."""
+		self.obj=NVDAObjectWithRole(role=controlTypes.ROLE_LISTITEM)
+		# Create and set a fake desktop object
+		self.fakeDesktopObj=NVDAObjectWithRole(role=controlTypes.ROLE_WINDOW)
+		api.setDesktopObject(self.fakeDesktopObj)
+		# Forcefully create a fake focus ancestry
+		globalVars.focusAncestors=[api.getDesktopObject(),NVDAObjectWithRole(role=controlTypes.ROLE_DIALOG),NVDAObjectWithRole(role=controlTypes.ROLE_LIST)]
+		braille.handler.handleGainFocus(self.obj)
+		# Make sure that we are testing with three regions

```

Comment:
It'd be super helpful if this comment could be expanded to list the roles (dialog, list, list item) and note that braille excludes the desktop object. I forgot about this last point myself and was thus very confused.